testing/regress/regress.md

```diff
@@ -0,0 +1,233 @@
+# Regression Testing
+
+## Why Regression Testing is Important
+
+Adding new features and making improvements is crucial for any product's evolution. However, these changes can sometimes inadvertently introduce regressions – previously fixed bugs that resurface or unexpected issues in existing functionality.
+
+The Regression Suite acts as a safety net, systematically verifying that core application functionality continues to work as expected after modifications. By running these tests, we can proactively identify and address regressions early in the development cycle, preventing them from reaching users and ensuring a stable experience. Think of it as a comprehensive health check for our software after any significant change.
+
+For official releases, the regression suite runs automatically as part of the build cycle. However, individual developers can (and should) use it when making significant changes, before issuing a Pull Request.
+
+## When to Use the Regression Suite
+
+Run the Regression Suite at these key points during the software development lifecycle:
+
+* **After each new build**: Following significant code changes, feature branch merges, or bug fixes, running the Regression Suite helps immediately detect any unintended side effects.
+* **Before a release candidate**: As we prepare a version for final testing and potential release, a thorough run of the Regression Suite is mandatory to confirm overall stability and identify any last-minute regressions.
```

Comment:
We don't currently do this.  We do run it on each release candidate after it is tagged.